### DISCUSSION TITLE
How to Stop MPU6050 DMP?

### DISCUSSION THREAD

**Discussion by williamesp2015** [jrowberg/i2cdevlib#596] [Q&A]
In my program I have to start and stop reading MPU6050 and I am using MPU6050_6Axis_MotionApps_V6_12. I couldn't find an reliable function to stop and restart DMP.

**Comment by flavioipiranga** [comment]
Hi @williamesp2015,

Check if this helps: https://github.com/jrowberg/i2cdevlib/blob/474c5e4fd2bbf3a486f40a79b7c908d0f49d3ab5/Arduino/MPU6050/MPU6050.cpp#L3022

**Comment by ZHomeSlice** [comment]
The DMP runs on the mpu6050 and fills the FIFO Buffer. this is all done in the background. this is required to keep the orientation correct. if the embedded DMP were to stop the detection of the change in direction would stop. 

options:
If you don't consistently empty the FIFO buffer will fill and overflow which corrupts the data within. so a simple reset `mpu.resetDMP();` clears the buffer and all you will need to do is wait for the next packet to arrive <= 10ms 
using the function  `if (mpu.dmpGetCurrentFIFOPacket(fifoBuffer)) { // Get the Latest packet`  will get that packed whenever you need it and it will handle overflows for you. so you don't need to use `mpu.resetDMP();` 

conclusion:
Run this function when you need to find out what orientation your bot is in whenever you need it.
`if (mpu.dmpGetCurrentFIFOPacket(fifoBuffer)) { `
and let the DMP keep track in the background for you.

Z
